### PR TITLE
executor: add `CopyReconstruct()` and `CopyConstruct()` methods to `Column`

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
Add  `CopyConstruct()` and `CopyReconstruct()` to `Column`, which is prepared for vectorized evaluation.

### What is changed and how it works?
1. Expose `CopyConstruct()` method;
2. Add a new method `CopyReconstruct()` method;

**Most of these changes are for unit tests, so it's easy to review.**

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: util/chunk/column.go [hunk 1]
```diff
@@ -63,15 +63,25 @@ type Column struct {
...
+	} else {
+		col = newFixedLenColumn(typeSize, cap)
 	}
-	return newFixedLenColumn(typeSize, cap)
+	col.ft = *ft
+	return col
+}
+
+// Type returns the type of data stored in this Column.
+func (c *Column) Type() *types.FieldType {
```
Comment (zz-jason) [inline util/chunk/column.go]: What is this method used for?

**Comment by qw4990** [comment]
A new method `CopyReconstruct(sel, dst)` is added, PTAL.

File: util/chunk/column.go (hunk 1 above)
Comment (coocood) [reply util/chunk/column.go]: I think we don't need to store FieldType in Column which makes the struct larger.

Comment (qw4990) [reply util/chunk/column.go]: The vectorized evaluation method for `Expression` may be defined as `VecEval(ctx Ctx, sel []int, result *Column)` and you can see there is no place to show what type of data the caller wants.
Therefore we add the `Type` method to `Column` to indicate what type of data should be in the `result`.

Comment (qw4990) [reply util/chunk/column.go]: If we remove this method, we have to implement a `VecEvalType` for each type or add a new argument to it like `VecEval(ctx, sel, result, type)`.
What's your opinion?

Comment (qw4990) [reply util/chunk/column.go]: Anyway, I removed this `Type()` method.
Let us discuss this problem in the next PR 🐒

Comment (qw4990) [reply util/chunk/column.go]: My fault, after discussing with @coocood , I think we don't need this method any more...

File: util/chunk/column.go [hunk 2]
```diff
@@ -96,7 +109,18 @@ func (c *Column) IsNull(rowIdx int) bool {
 	return nullByte&(1<<(uint(rowIdx)&7)) == 0
 }
 
-func (c *Column) copyConstruct() *Column {
+// CopyConstruct copies this Column to dst.
+// If dst is nil, it creates a new Column and returns it.
+func (c *Column) CopyConstruct(dst *Column) *Column {
+	if dst != nil {
```
Comment (coocood) [inline util/chunk/column.go]: This can save a few lines.

```
If dst == nil {
    dst = &Column{length: c.length, nullCount: c.nullCount}
}
```

Comment (qw4990) [reply util/chunk/column.go]: But it will cause more code changes...

**Comment by coocood** [comment]
LGTM

**Pull request merged**